<a href="https://colab.research.google.com/github/Mataf777/ClassColab/blob/Proyecto-1/Proyecto1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Proyecto etapa 1: ADD

#Exploracion Metricas MLB

1. Contexto y objetivo
Este análisis se centra en la exploración descriptiva de métricas avanzadas de rendimiento ofensivo en jugadores de MLB, evaluando variables relacionadas con la calidad del contacto de bateo y la disciplina de bateo. El objetivo es caracterizar estadísticamente el conjunto de datos, analizar la distribución de sus variables y explorar la relación entre métricas de datos de jugadores de la liga MLB temporada 2025 y 2024.

2. Problema que se desea estudiar.
Estudiar si los jugadores que conectan la pelota con mayor fuerza y precisión tienden a tener mejores resultados ofensivos. Para ello, se analizarán métricas de calidad de contacto bateo (hard_hit%, barrel%, sweet_spot%) y métricas ofensivas avanzadas (wOBA, xwOBA) y rendimiento de jugadores.

3. Justificación de su relevancia.
Comprender estas relaciones permite:
Identificar patrones en el rendimiento, ya que seria fundamental para aplicaciones en evaluación deportiva.
Evaluar coherencia entre métricas esperadas (xwOBA) y reales (wOBA).
Fundamentar análisis posteriores predictivos.


5. Analisis de estructura


In [15]:
import pandas as pd
import pandas as pd
pd.set_option('display.float_format', '{:.3f}'.format)

df = pd.read_csv("/content/drive/MyDrive/ClassFiles/beisstats.csv")

df.shape
df.info()
df.isnull().sum()
df.describe()



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253 entries, 0 to 252
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   last_name, first_name  253 non-null    object 
 1   player_id              253 non-null    int64  
 2   year                   253 non-null    int64  
 3   pa                     253 non-null    int64  
 4   k_percent              253 non-null    float64
 5   bb_percent             253 non-null    float64
 6   woba                   253 non-null    float64
 7   xwoba                  253 non-null    float64
 8   sweet_spot_percent     253 non-null    float64
 9   barrel_batted_rate     253 non-null    float64
 10  hard_hit_percent       253 non-null    float64
 11  avg_best_speed         253 non-null    float64
 12  avg_hyper_speed        253 non-null    float64
 13  whiff_percent          253 non-null    float64
 14  swing_percent          253 non-null    float64
dtypes: flo

player_id     year      pa  k_percent  bb_percent    woba   xwoba  \
count    253.000  253.000 253.000    253.000     253.000 253.000 253.000   
mean  636267.154 2024.502 635.043     22.076       7.381   0.308   0.315   
std    54641.226    0.501 109.254      4.442       1.855   0.030   0.026   
min   434378.000 2024.000 408.000     11.800       2.800   0.226   0.247   
25%   607192.000 2024.000 543.000     19.100       6.100   0.288   0.300   
50%   657006.000 2025.000 655.000     21.400       7.400   0.309   0.316   
75%   671737.000 2025.000 722.000     24.800       8.800   0.325   0.333   
max   808967.000 2025.000 856.000     35.100      12.300   0.406   0.395   

       sweet_spot_percent  barrel_batted_rate  hard_hit_percent  \
count             253.000             253.000           253.000   
mean               34.122               8.331            40.165   
std                 2.921               1.714             4.078   
min                24.400               3.300            28.900   
25%                32.400               7.200            37.400   
50%                34.500               8.300            40.200   
75%                36.300               9.500            42.700   
max                41.800              12.700            50.000   

       avg_best_speed  avg_hyper_speed  whiff_percent  swing_percent  
count         253.000          253.000        253.000        253.000  
mean           78.289           94.297         24.380         48.291  
std             1.796            0.623          3.938          2.502  
min            73.587           92.672         15.700         42.500  
25%            77.149           93.900         21.600         46.400  
50%            78.394           94.323         24.000         48.300  
75%            79.629           94.724         27.300         50.000  
max            82.826           95.946         37.700         55.700

4. Preguntas Descriptivas

     1.-Cual es la distribucion del wOBA (promedio de embase) entre los jugadores?


In [10]:
import numpy as np

print("Media:", round(df['woba'].mean(), 3))
print("Mediana:", np.median(df['woba']))
print("Desviación estándar:", round(df['woba'].std(), 3))
print("Mínimo:", np.min(df['woba']))
print("Máximo:", np.max(df['woba']))


Media: 0.308
Mediana: 0.309
Desviación estándar: 0.03
Mínimo: 0.226
Máximo: 0.406


El wOBA(Promedio para embasarse ponderado) presenta una media de 0.308 y una mediana muy cercana, lo que indica una distribución simétrica y un rendimiento ofensivo estable entre los jugadores analizados. La desviación estándar moderada sugiere que, aunque el grupo es relativamente homogéneo, existen diferencias individuales relevantes que podrían explicarse por variaciones en la calidad de contacto o disciplina en el plato.

 2.-Cual es la media y desviacion estandar de hard_hit% (pelotas bateadas con velocidad de salida alta)



In [14]:
print("Media hard_hit%:", round(df['hard_hit_percent'].mean(), 3))
print("Desviación estándar hard_hit%:", round(df['hard_hit_percent'].std(), 3))
print("Mínimo:", df['hard_hit_percent'].min())
print("Máximo:", df['hard_hit_percent'].max())



Media hard_hit%: 40.165
Desviación estándar hard_hit%: 4.078
Mínimo: 28.9
Máximo: 50.0


El hard_hit% presenta una media de 40.17%, lo que indica que el grupo analizado genera contacto fuerte con una frecuencia superior al promedio general de la liga. La desviación estándar moderada (4.08) sugiere una variabilidad controlada entre jugadores, aunque se observan diferencias importantes entre el valor mínimo (28.9%) y el máximo (50%), lo que indica que la capacidad de generar contacto fuerte puede ser un factor diferenciador clave en el rendimiento ofensivo.

 3.-Existe correlacion entre barrel%(batazo con potencial) y xwOBA (embase esperado)?



In [18]:
from scipy.stats import pearsonr


r, p = pearsonr(df['barrel_batted_rate'], df['xwoba'])
print(f"Correlación Barrel% vs xwOBA: {r:.3f}")
print(f"p-value: {p:.3e}")


Correlación Barrel% vs xwOBA: 0.498
p-value: 2.929e-17


Se encontró una correlación positiva moderada (r = 0.498) entre barrel%(calidad de contacto del bateador) y xwOBA(Promedio de Embasarse Ponderado esperado), la cual es estadísticamente significativa (p < 0.001). Esto indica que los jugadores que generan una mayor proporción de batazos con alto potencial tienden a presentar mejores valores de rendimiento ofensivo esperado. Aproximadamente el 25% de la variabilidad del xwOBA puede explicarse por el barrel%, lo que confirma que la calidad del contacto es un factor relevante en la producción ofensiva

 4.-Que jugadores presentan mayor whiff% (abanicar ) y como se relaciona con k_percent (strikes)?



In [17]:
df.sort_values(by='whiff_percent', ascending=False)[
    ['last_name, first_name','whiff_percent','k_percent']
].head(5)

from scipy.stats import pearsonr

r, p = pearsonr(df['whiff_percent'], df['k_percent'])
print(f"Correlacion Whiff vs K: {r:.3f}")
print(f"p-value: {p:.3e}")




Correlacion Whiff vs K: 0.866
p-value: 1.531e-77


Se encontró una correlación positiva muy fuerte entre whiff%(bateador abanica y falla ) y K%(ponches) (r = 0.866), estadísticamente significativa (p < 0.001). Esto indica que los jugadores que presentan mayores tasas de swings fallidos tienden a registrar significativamente más ponches. Aproximadamente el 75% de la variabilidad en el K% puede explicarse por el whiff%, lo que confirma que la frecuencia de abanicos es un indicador determinante del riesgo ofensivo.


 5.-Existe diferencia significative entre wOBA y xwOBA en promedio?


In [19]:
from scipy.stats import ttest_rel

t_stat, p_value = ttest_rel(df['woba'], df['xwoba'])

print(f"Estadístico t: {t_stat:.3f}")
print(f"p-value: {p_value:.3e}")


Estadístico t: -7.587
p-value: 6.323e-13


La prueba *paired T test* mostró una diferencia estadísticamente significativa entre wOBA(promedio para embasarse ponderado) y xwOBA(Promedio de Embasarse Ponderado esperado) (t = -7.59, p < 0.001), lo que indica que el rendimiento ofensivo observado difiere del esperado según la calidad del contacto. El signo negativo del estadístico sugiere que, en promedio, los jugadores presentan un wOBA inferior a su xwOBA, lo que podría reflejar influencia de factores externos como variabilidad, defensa o efectos situacionales.

# Conclusion final

El análisis muestra que el rendimiento ofensivo del grupo es relativamente estable, aunque con diferencias individuales relevantes. La calidad del contacto, especialmente el barrel%, se relaciona de forma significativa con el rendimiento ofensivo esperado, mientras que el whiff% explica fuertemente la tasa de ponches. Además, la diferencia estadísticamente significativa entre wOBA y xwOBA indica que el desempeño real no depende únicamente de la calidad del contacto, sino también de factores externos. En conjunto, los resultados confirman que las métricas avanzadas permiten comprender mejor los determinantes del rendimiento ofensivo en el béisbol profesional.